android/java/org/chromium/chrome/browser/ntp/BraveNewTabPageLayout.java

```diff
@@ -732,6 +732,10 @@ private void refreshFeed() {
 
     private void keepPosition(int prevScrollPosition, int prevRecyclerViewPosition,
             int prevRecyclerViewItemPosition) {
+        if ((!mIsNewsOn && mIsShowNewsOn) || (mIsNewsOn && mIsShowOptin)
```

Comment:
Not subject or request to change, but the fact which may be useful to understand the intention:

`(!mIsNewsOn && mIsShowNewsOn) || (mIsNewsOn && !mIsShowNewsOn)`
is equal algebraically  to just 
`(mIsNewsOn != mIsShowNewsOn)`